### ISSUE TITLE
[BUG]OperatorHub not accessible via proxy.

### ISSUE THREAD

**Issue by vyom-soft** [code-ready/crc#1956]
I have read this **https://github.com/code-ready/crc/issues/898** but still the problem is there. Could you please see once.

### General information

  * OS:  **Windows10**
  * Hypervisor: **Hyper-V** 
  * Did you run `crc setup` before starting it (Yes/No)? Yes
  * Running CRC on:  VM

## CRC version
**4.6.6**

## CRC status
[crc-4.6.6-log.txt](https://github.com/code-ready/crc/files/5931436/crc-4.6.6-log.txt)
DEBU HTTP-PROXY: http://my-proxy.company.com:3128, HTTPS-PROXY: http://my-proxy.company.com:3128, 
NO-PROXY: 127.0.0.1,localhost,127.0.0.1,.testing,.*, 
proxyCAFile: C:/Users/badam56/crc/crc/certificate_chain.pem
DEBU CodeReady Containers version: 1.20.0+ef3f80d
DEBU OpenShift version: 4.6.6 (embedded in executable)
Checking file: C:\Users\badam56\.crc\machines\crc\.crc-exist
DEBU Copying 'C:\Users\badam56\.crc\cache\crc_hyperv_4.6.6\oc.exe' to 'C:\Users\badam56\.crc\bin\oc\oc.exe'
DEBU About to run SSH command:
df -B1 --output=size,used,target /sysroot | tail -1
DEBU SSH command results: err: <nil>, output: 32720793600 14988636160 /sysroot
DEBU About to run SSH command with hidden output
DEBU SSH command succeeded
CRC VM:          Running
OpenShift:       Running (v4.6.6)
Disk Usage:      14.99GB of 32.72GB (Inside the CRC VM)
Cache Usage:     15.54GB
Cache Directory: C:\Users\badam56\\.crc\cache

```

## CRC config
- disable-update-check           : true
- http-proxy                            : http://my-proxy.company.com
[oc-get-proxy-oyaml.txt](https://github.com/code-ready/crc/files/5931469/oc-get-proxy-oyaml.txt)
:3128
- https-proxy                           : http://my-proxy.company.com:3128
- memory                                : 16000
- nameserver                            : 29.136.0.100
- no-proxy                              : localhost,127.0.0.1,.testing
- proxy-ca-file                         : C:/Users/badam56/crc/crc/certificate_chain.pem
- pull-secret-file                      : pull-secret.txt
- skip-check-goodhosts-cached           : true

```

## Host Operating System
 **Microsoft Windows 10 Enterprise LTSC**

### Logs

**$ oc get pods -n openshift-marketplace**
NAME                                    READY   STATUS             RESTARTS   AGE
certified-operators-5w7m9               0/1     ImagePullBackOff   0          63d
certified-operators-bl9md               0/1     ImagePullBackOff   0          17h
community-operators-hml4l               0/1     ImagePullBackOff   0          62d
community-operators-hv72f               0/1     ImagePullBackOff   0          17h
marketplace-operator-74d997db66-88vvk   1/1     Running            0          16h
redhat-marketplace-kb4gk                0/1     ImagePullBackOff   0          63d
redhat-marketplace-kvhw2                0/1     ImagePullBackOff   0          17h
redhat-operators-gnchq                  0/1     ImagePullBackOff   0          17h
redhat-operators-jsc9l                  0/1     ImagePullBackOff   0          63d

**$ oc get opsrc -n openshift-marketplace**
error: the server doesn't have a resource type "opsrc"

**Comment by praveenkumar** [comment]
@vyom-soft looks like you are using 1.20.0 version of crc, with 1.21.1 we have improved code around proxy, can you try newer version and let us know if still the same issue for you?

**Comment by vyom-soft** [comment]
@praveenkumar Thank you for seeing it.
Infact I tried with 1.21.1. But is does not start. Buring setup it threw the below error.  **Hence I reverted back to 1.20.0**

**"Failed to update cluster proxy configuration: Failed to add proxy cert details ssh command error:
command : oc apply -f /tmp/user-ca-bundle.json --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 1\n: error: error parsing /tmp/user-ca-bundle.json: json: line 3: invalid character '\r' in string literal"**

**Comment by praveenkumar** [comment]
@vyom-soft when you tried `1.21.1` , did you run the `crc cleanup` also?

> Also there is new " admin-helper-windows.exe" Please can you inform its purpose, what it is going to do.

This is just renamed of the `goodhosts` binary as of now, we have other plans around it.

**Comment by praveenkumar** [comment]
> "Failed to update cluster proxy configuration: Failed to add proxy cert details ssh command error:
command : oc apply -f /tmp/user-ca-bundle.json --context admin --cluster crc --kubeconfig /opt/kubeconfig
err : Process exited with status 1\n: error: error parsing /tmp/user-ca-bundle.json: json: line 3: invalid character '\r' in string literal"

@vyom-soft This looks like your ca bundle json is not valid one, please make sure you are using the valid one, looks like since you are on windows so might be the it adds `\r` for end of line char. when adding using the cert.

**Comment by vyom-soft** [comment]
> @vyom-soft when you tried `1.21.1` , did you run the `crc cleanup` also?
>  **No**
> > Also there is new " admin-helper-windows.exe" Please can you inform its purpose, what it is going to do.
> 
> This is just renamed of the `goodhosts` binary as of now, we have other plans around it.
Thank you for the info

**Comment by vyom-soft** [comment]
> > "Failed to update cluster proxy configuration: Failed to add proxy cert details ssh command error:
> > command : oc apply -f /tmp/user-ca-bundle.json --context admin --cluster crc --kubeconfig /opt/kubeconfig
> > err : Process exited with status 1\n: error: error parsing /tmp/user-ca-bundle.json: json: line 3: invalid character '\r' in string literal"
> 
> @vyom-soft This looks like your ca bundle json is not valid one, please make sure you are using the valid one, looks like since you are on windows so might be the it adds `\r` for end of line char. when adding using the cert.

Okay, I am not sure If I can give a try soon again with 1.21.1.  But the same cert is feeded to 4.6.6

**Comment by praveenkumar** [comment]
> Okay, I am not sure If I can give a try soon again with 1.21.1. But the same cert is feeded to 1.20.0 (4.6.6)

@vyom-soft with 1.20.0 there is some really breaking things for proxy, It would be really great to try 1.21.1 

Also try to open the bundle file in notepad ++ or some other tools to remove the windows specific carriage return.

**Comment by vyom-soft** [comment]
> > Okay, I am not sure If I can give a try soon again with 1.21.1. But the same cert is feeded to 1.20.0 (4.6.6)
> 
> @vyom-soft with 1.20.0 there is some really breaking things for proxy, It would be really great to try 1.21.1
> 
> Also try to open the bundle file in notepad ++ or some other tools to remove the windows specific carriage return.

@praveenkumar  thank you for the support. In Notpad++, I always have Unix Line feed character setting. 
I will see when can I give a try 1.21.1.

**Comment by praveenkumar** [comment]
@vyom-soft Hi, we also released 1.22.0 recently can you give that a try if you already not tried 1.21.0 and let us know if you still have issue with proxy?

**Comment by vyom-soft** [comment]
> @vyom-soft Hi, we also released 1.22.0 recently can you give that a try if you already not tried 1.21.0 and let us know if you still have issue with proxy?

Thank you very much. I can give a try on Monday.

**Comment by vyom-soft** [comment]
@praveenkumar, Thank you. After installing version v1.22.0 and configuring the proxy, The OperatorHub is accessible again.
You can close this.
